sympy/integrals/prde.py

```diff
@@ -1038,6 +1049,47 @@ def is_deriv_k(fa, fd, DE):
             return (ans, result, const)
 
 
+def base_Q(G):
+    """
+    >>> G = [Poly(x**2*sqrt(2) + x, x, extension=True),
+            Poly(x**2*sqrt(3) + 1, x, extension=True)]
+    >>> base_Q(G)
+    ((Rational function field in  over QQ<sqrt(2) + sqrt(3)> with lex order,
+      [ANP([1/2, 0, -9/2, 0], [1, 0, -10, 0, 1], QQ)/ANP([1], [1, 0, -10, 0, 1], QQ),
+          -ANP([-1/2, 0, 11/2, 0], [1, 0, -10, 0, 1], QQ)/ANP([1], [1, 0, -10, 0, 1], QQ)]),
+     QQ<sqrt(2) + sqrt(3)>)
+
+    """
+    from sympy import sfield, AlgebraicField, QQ
+    # algebracic constants
+    algebraic_consts = set()
+    # transcendental constants that are algebraically
+    # independent
+    alg_ind_trans_consts = set()
+
+    for poly in G:
+        trans_consts = list(set(poly.gens) - set([poly.gen]))
```

Comment:
> I wonder why all generators except for the principal one (poly.gen) are taken.

Here "all generators" -> "transcendental generators" ?
As for not including `poly.gen`, I think you are right, I should include it. But I think I haven't found a good enough test case for this function. I'll try to do that first and see if it works okay.